src/dodal/devices/i10/i10_apple2.py

```diff
@@ -299,10 +323,77 @@ async def set(self, value: SupportsFloat) -> None:
                 f"jaw_phase position for angle ({value}) is outside permitted range"
                 f" [-{self.jaw_phase_limit}, {self.jaw_phase_limit}]"
             )
-        await self.id_ref().id_jaw_phase().set(jaw_phase)
+        await self.id_ref().id_jaw_phase.set(jaw_phase)
         self._angle_set(value)
 
 
+class I10Id(Device):
+    def __init__(
+        self,
+        pgm: PGM,
+        prefix: str,
+        look_up_table_dir: str,
+        source: tuple[str, str],
+        jaw_phase_limit=12.0,
+        jaw_phase_poly_param=DEFAULT_JAW_PHASE_POLY_PARAMS,
+        angle_threshold_deg=30.0,
+        name: str = "",
+    ) -> None:
+        """A compound device to make up the full I10 insertion device.
+         This is in effect a single I10Apple2 with three different set methods for
+         energy, polarisation and linear arbitrary angle. See
+         `UML </_images/i10_id_design.png>`__ for detail.
+        .. figure:: /explanations/umls/i10_id_design.png
+        Attributes
+        ----------
+            energy: I10Apple2PGM
+                Devices that move both pgm and id energy at the same time.
+            pol: I10Apple2Pol
+                Devices that control the x-ray polarisation.
+            laa: LinearArbitraryAngle
+                Devices that allow alteration of the beam polarisation angle in LA mode.
+        """
+
+        self.energy = I10Apple2PGM(
+            id=I10Apple2(
+                id_gap=UndulatorGap(name="id_gap", prefix=prefix),
+                id_phase=UndulatorPhaseAxes(
+                    name="id_phase",
+                    prefix=prefix,
+                    top_outer="RPQ1",
+                    top_inner="RPQ2",
+                    btm_inner="RPQ3",
+                    btm_outer="RPQ4",
+                ),
+                id_jaw_phase=UndulatorJawPhase(
+                    prefix=prefix,
+                    move_pv="RPQ1",
+                ),
+                energy_gap_table_path=Path(
+                    look_up_table_dir + "IDEnergy2GapCalibrations.csv",
+                ),
+                energy_phase_table_path=Path(
+                    look_up_table_dir + "IDEnergy2PhaseCalibrations.csv",
+                ),
+                name="idd_energy",
+                source=source,
+                prefix="",
+            ),
+            pgm=pgm,
+            name="energy",
+        )
```

Comment:
Should: It's weird to me that the inner objects are instantiated here. I would expect this instead to just be something like:

```suggestion
        self._id = I10Apple2(prefix, look_up_table_dir, source)
        self.energy = I10Apple2PGM(id=I10Apple2(self._id, pgm, name="energy")
```

and the rest to be instantiated in the constructor of `I10Apple2`